sources/xfscli/src/bin/fsck_xfs.rs

```diff
@@ -0,0 +1,169 @@
+            let mut repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+
+            // Clear the log, mount and unmount the XFS file system
+            // before xfs_repair
+            if repair_exit_code == 2 && repair {
+                println!("Replaying log for {}", target);
+                let mkdir_result = fs::create_dir_all("/tmp/repair_mnt");
+                if mkdir_result.is_err() {
+                    exit(1);
+                }
+
+                let params =
+                    fs::read_to_string("/proc/cmdline").expect("Unable to read /proc/cmdline file");
+                let mut root = String::new();
+                let mut root_flags = "-o ".to_string();
+                for param in params.split(' ') {
+                    if param.starts_with("root=") {
+                        root = param.strip_prefix("root=").unwrap().to_string();
+                    }
+                    if param.starts_with("rootflags=") {
+                        root_flags = root_flags
+                            .clone()
+                            .add(param.strip_prefix("rootflags=").unwrap());
+                    }
+                }
+                let path = Path::new(&root);
+                let metadata = path.metadata().expect("Could not fetch metadata");
+                let file_type = metadata.file_type();
+
+                if !file_type.is_block_device() {
+                    let blkid_result = Command::new(BLKID)
+                        .args(["-t", &root, "-o", target])
+                        .output()
+                        .expect("Couldn't run blkid");
+                    root = String::from_utf8(blkid_result.stdout).unwrap();
+                }
+
+                let basename_dev = Path::new(target).file_name().unwrap();
+
+                let basename_root = Path::new(&root).file_name().unwrap();
+
+                let mut mount_args = vec![target.as_str(), "/tmp/repair_mnt"];
+                if basename_dev == basename_root {
+                    mount_args.push(root_flags.as_str());
+                }
+
+                let mount_result = Command::new(MOUNT)
+                    .args(mount_args)
+                    .output()
+                    .expect("Couldn't run mount");
+                if !mount_result.status.success() {
+                    exit(1);
+                }
+
+                Command::new(UMOUNT)
+                    .args(["/tmp/repair_mnt"])
+                    .output()
+                    .expect("Couldn't run unmount");
+
+                println!("Performing XFS repair again after cleaning the log..");
+                repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+                fs::remove_dir("/tmp/repair_mnt")
+                    .expect("could not delete the /tmp/repair_mnt dir");
+            }
+            exit(repair2fsck_code(repair_exit_code));
+        }
+
+        if auto {
+            println!("fsck.xfs : XFS file system.");
```

Comment:
Does `auto` basically mean `NOOP`? Or why is this print required?

Reply:
Yes, in fsck.xfs with auto=true, means NOOP. This print is required to replicate the behavior in  xfs shell [script](https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/fsck/xfs_fsck.sh)